backend/database/versions/26_e08b6b67f076_make_collection_id_pkey.py

```diff
@@ -0,0 +1,126 @@
+"""make_collection_id_pkey
+
+Revision ID: 25_e08b6b67f076
+Revises: 25_43e20ba9f9f0
+Create Date: 2022-03-04 21:54:51.795657
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "26_e08b6b67f076"
+down_revision = "25_43e20ba9f9f0"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # Drop foreign keys that reference composite key [project.id, project.visibility]
+    op.drop_constraint("dataset_collection_id_fkey", "dataset", type_="foreignkey")
+    op.drop_column("dataset", "collection_visibility")
+    op.drop_constraint("project_link_collection_id_fkey", "project_link", type_="foreignkey")
+    op.drop_column("project_link", "collection_visibility")
+
+    # Drop foreign key that references composite key [project.id, project.visibility]
+    op.drop_constraint("geneset_collection_id_fkey", "geneset", type_="foreignkey")
+    # Drop unique key that uses collection_visibility
+    op.drop_constraint("_geneset_name__collection_uc", "geneset", type_="unique")
+    op.drop_column("geneset", "collection_visibility")
+    # Recreate unique key without collection_visibility
+    op.create_unique_constraint(op.f("_geneset_name__collection_uc"), "geneset", ["name", "collection_id"])
+
+    # Add project.revision_of
+    op.add_column("project", sa.Column("revision_of", sa.String(), nullable=True))
+
+    # Drop project composite primary key; add project.id as primary key
+    op.drop_constraint("project_pkey", "project")
+    op.create_primary_key(op.f("pk_project"), "project", ["id"])
+
+    # Add foreign keys for project.id
+    op.create_foreign_key(op.f("fk_dataset_collection_id_project"), "dataset", "project", ["collection_id"], ["id"])
```

Comment:
Is it required to have so many lists with a single value?

Reply:
yeah it's the way this function accepts column arguments in order to accommodate composites